### PR TITLE
RestaurantForm's google api integration

### PR DESCRIPTION
**Pull request by RoniNiklas**
- Typing in namefield fetches autocomplete suggestions (after 1 second timeout)
- Clicking on a suggestion fetches and populates the restaurant's details into the form.
- Changing the value in the name field undefines the placeId.
- Changing the address requires a new address check
- Restaurants can be posted without url, but if a url is provided it needs to be between 3 and 240 characters.
- Submit button is enabled by default when editing a restaurant (editing the address needs a new address check, editing the name undefines placeId).
- New Error checking instead of form-hook
- Check button moved to the address field
- Sending a restaurant suggestion / adding a restaurant adds placeId if it has been fetched

### CODE REVIEW

File: packages/app/src/components/Restaurants/AutocompleteEntry/AutocompleteEntry.css [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.list-group-item {
+    color:black;
+    padding: 0.5em 1em;
+}
+
+.list-group-item:hover {
+    transform: scale(1.025);
+    cursor: pointer;
+}
```
Comment (ritinae) [inline packages/app/src/components/Restaurants/AutocompleteEntry/AutocompleteEntry.css]: T R I G G E R E D

File: packages/app/src/components/Restaurants/AutocompleteList/AutocompleteList.css [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.autocompletelist {
+    margin-bottom: 1em;
+    margin-top: 0em;
+}
```
Comment (ritinae) [inline packages/app/src/components/Restaurants/AutocompleteList/AutocompleteList.css]: T R I G G E R E D

File: packages/app/src/components/Restaurants/form/RestaurantForm.css [hunk 3]
```diff
@@ -22,3 +22,12 @@
...
 }
+
+.name-field:focus {
+    margin-bottom: 0em;
+}
+
+.address-field {
+    display:flex;
+    flex-direction: row;
+}
```
Comment (ritinae) [inline packages/app/src/components/Restaurants/form/RestaurantForm.css]: T R I G G E R E D

File: packages/app/src/components/Restaurants/form/RestaurantForm.js [hunk 4]
```diff
@@ -1,90 +1,161 @@
...
+
+  const handleNameChange = (name) => {
+    setRestaurant({
+      ...restaurant,
+      name,
+      placeId: undefined
+    })
+    setErrors({ ...errors, name: undefined })
+    clearTimeout(timeoutHandle)
+    setTimeoutHandle((setTimeout(async () => {
```
Comment (ritinae) [inline packages/app/src/components/Restaurants/form/RestaurantForm.js]: The `async` here is unnecessary as `await` is not used within the callback?

File: packages/app/src/components/Restaurants/form/RestaurantForm.js [hunk 5]
```diff
@@ -188,13 +264,13 @@ RestaurantForm.propTypes = {
...
+    })
   }),
   onSubmit: PropTypes.func,
   setRestaurant: PropTypes.func.isRequired,
   suggestTooltip: PropTypes.string,
   submitMessage: PropTypes.string,
+  autocompleteDelay: PropTypes.number
 }
 
 export default RestaurantForm
```
Comment (ritinae) [inline packages/app/src/components/Restaurants/form/RestaurantForm.js:277]: This component is starting to get bloated. Might want to refactor things to smaller components and hooks later on. This is ok for now, though.

File: packages/app/src/components/RouteMap/RouteMap.js [hunk 6]
```diff
@@ -11,14 +11,16 @@ const RouteMap = ({ restaurant }) => {
...
     const getMapData = async () => {
       const leg = await locationService.getLeg(restaurant.coordinates.latitude, restaurant.coordinates.longitude)
-      setStart([locationService.unityLat, locationService.unityLon])
-      setEnd([restaurant.coordinates.latitude, restaurant.coordinates.longitude])
-      setBounds(locationService.calculateBounds(leg))
-      setPositions(locationService.decodeRoute(leg.legGeometry.points))
+      mounted && setStart([locationService.unityLat, locationService.unityLon])
+      mounted && setEnd([restaurant.coordinates.latitude, restaurant.coordinates.longitude])
+      mounted && setBounds(locationService.calculateBounds(leg))
+      mounted && setPositions(locationService.decodeRoute(leg.legGeometry.points))
```
Comment (ritinae) [inline packages/app/src/components/RouteMap/RouteMap.js]: why not just wrap this to an `if`? This looks kinda iffy

File: packages/app/src/components/Restaurants/AutocompleteEntry/AutocompleteEntry.css (hunk 1 above)
Comment (aLahdekorpi) [inline packages/app/src/components/Restaurants/AutocompleteEntry/AutocompleteEntry.css:9]: tyhjä perään

Comment (aLahdekorpi) [inline packages/app/src/components/Restaurants/AutocompleteEntry/AutocompleteEntry.css:9]: empty line

File: packages/app/src/components/Restaurants/AutocompleteEntry/AutocompleteEntry.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+AutocompleteEntry.propTypes = {
+  suggestion: PropTypes.shape({
+    placeId: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
+    address: PropTypes.string.isRequired
+  }),
+  handleClick: PropTypes.func.isRequired
+}
+
+export default AutocompleteEntry
```
Comment (aLahdekorpi) [inline packages/app/src/components/Restaurants/AutocompleteEntry/AutocompleteEntry.js:27]: -

File: packages/app/src/components/Restaurants/AutocompleteList/AutocompleteList.css (hunk 2 above)
Comment (aLahdekorpi) [inline packages/app/src/components/Restaurants/AutocompleteList/AutocompleteList.css:4]: -

File: packages/app/src/components/Restaurants/AutocompleteList/AutocompleteList.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    </Card>
+  )
+}
+
+AutocompleteList.propTypes = {
+  suggestions: PropTypes.array.isRequired,
+  handleClick: PropTypes.func.isRequired
+}
+
+export default AutocompleteList
```
Comment (aLahdekorpi) [inline packages/app/src/components/Restaurants/AutocompleteList/AutocompleteList.js:22]: -

File: packages/app/src/components/Restaurants/EditForm.test.js [hunk 9]
```diff
@@ -73,20 +82,22 @@ describe('when logged in', () => {
...
 
     restaurantService.getOneById.mockResolvedValue(
       {
         name: 'Luigi\'s pizza',
+        address: 'kotikatu 1 a',
         url: 'www.pizza.fi',
         id: 1,
         categories: [],
-        coordinates: {latitude: 60, longitude: 24}
+        coordinates: { latitude: 60, longitude: 24 }
```
Comment (aLahdekorpi) [inline packages/app/src/components/Restaurants/EditForm.test.js:100]: The same restaurant copy pasted to a lot of tests.

File: packages/app/src/components/Restaurants/form/RestaurantForm.css (hunk 3 above)
Comment (aLahdekorpi) [inline packages/app/src/components/Restaurants/form/RestaurantForm.css:33]: empty line

**Review by aLahdekorpi — COMMENTED** [review]
- EditForm.test.js has a lot of copy-paste. Maybe look into it if possible. 
- Otherwise well done

**Review by skoskipaa — APPROVED** [review]

**Pull request merged**